infra/modules/azure_api_management/monitor.tf

```diff
@@ -0,0 +1,153 @@
+resource "azurerm_api_management_logger" "this" {
+  count = var.application_insights.enabled && var.tier != "test" ? 1 : 0
+
+  name                = "${local.apim.name}-logger"
+  api_management_name = azurerm_api_management.this.name
+  resource_group_name = var.resource_group_name
+
+  dynamic "application_insights" {
+    for_each = var.management_logger_application_insight_enabled ? [1] : []
+    content {
+      instrumentation_key = var.application_insights.instrumentation_key
+    }
+  }
+
+}
+
+resource "azurerm_api_management_diagnostic" "this" {
```

Comment:
Mmh what's the purpose of this?
I'm asking because it seems to be a general settings for all the API groups, but it is likely we will need granular control over each API group. Even though this could be useful later on when we will agree on APIM best practices and determine if APIM will be the API gateway in front of all.
Don't know, me might also remove it from the module and eventually add it on the client side, as we will always have a single APIM instance

Reply:
Ok so let's remove it from now, we'll understand later what to do